autogpt_platform/backend/backend/util/workspace_transfer_test.py

```
@@ -0,0 +1,400 @@
+    mock_db.get_workspace_file_by_path.return_value = occupant
+    mock_db.get_workspace_total_size.return_value = 2500
+    mock_db.create_workspace_file.return_value = _make_workspace_file(
+        id="new-copy", path="/reports/summary.pdf"
+    )
+
+    # 2500 + 1000 = 3500 > 3000 would fail; freeing the 1000-byte occupant
+    # drops usage to 1500, so 1500 + 1000 = 2500 fits.
+    with _patched(mock_db, mock_storage, storage_limit=3000):
+        with patch.object(
+            WorkspaceManager, "delete_file", AsyncMock(return_value=True)
+        ):
```

Ruff (0.16.0)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)